NewHorizons/INewHorizons.cs

```diff
@@ -222,5 +237,17 @@ AudioSignal SpawnSignal(IModBehaviour mod, GameObject root, string audio, string
         /// </summary>
         /// <param name="id"></param>
         void SetNextSpawnID(string id);
+
+        /// <summary>
+        /// Registers a builder for the main menu.
+        /// Call this once before the main menu finishes loading
+        /// </summary>
+        /// <param name="mod"></param>
+        /// <param name="builder">Builder to run when this title screen is selected.</param>
+        /// <param name="disableNHPlanets">If set to true, NH generated planets will not show on the title screen. If false, this title screen has the same chance as other NH planet title screens to show.</param>
+        /// <param name="shareTitleScreen">If set to true, this custom title screen will merge with all other custom title screens with shareTitleScreen set to true. If false, NH will randomly select between this and other valid title screens that are loaded.</param>
+        /// <param name="conditionRequired">Persistent condition required for this title screen to appear.</param>
+        /// <param name="factRequired">Ship log fact required for this title screen to appear.</param>
+        void RegisterTitleScreenBuilder(IModBehaviour mod, Action<GameObject> builder, bool disableNHPlanets = true, bool shareTitleScreen = false, string conditionRequired = null, string factRequired = null);
```

Comment:
Why does the builder have to take in a game object? For example Codec when it handles the title screen calls a method with no parameters. Might be simplest for mods to integrate if the method takes in nothing. Otherwise at least add a comment saying what game object its passing in to the method and users can just choose to not use the parameter

Reply:
I'll add a comment